src/NServiceBus.Core/Pipeline/Outgoing/AttachSenderRelatedInfoOnMessageBehavior.cs

```diff
@@ -17,7 +19,30 @@ public Task Invoke(IRoutingContext context, Func<IRoutingContext, Task> next)
 
             if (!message.Headers.ContainsKey(Headers.TimeSent))
             {
-                message.Headers[Headers.TimeSent] = DateTimeOffsetHelper.ToWireFormattedString(DateTimeOffset.UtcNow);
+                message.Headers[Headers.TimeSent] = DateTimeOffsetHelper.ToWireFormattedString(utcNow);
+            }
+
+            if (!message.Headers.ContainsKey(Headers.DeliverAt))
+            {
+                if (context.Extensions.TryGet<DispatchProperties>(out var dispatchProperties))
+                {
+                    if (dispatchProperties.DelayDeliveryWith != null)
+                    {
+                        var timeDelay = dispatchProperties.DelayDeliveryWith.Delay;
+                        if (DateTimeOffset.TryParse(message.Headers[Headers.TimeSent], out var timeSent))
+                        {
+                            message.Headers[Headers.DeliverAt] = DateTimeOffsetHelper.ToWireFormattedString(timeSent.Add(timeDelay));
+                        }
+                        else
+                        {
+                            message.Headers[Headers.DeliverAt] = DateTimeOffsetHelper.ToWireFormattedString(utcNow.Add(timeDelay));
+                        }
```

Comment:
Don't think it's worth trying to parse, as this is only designed for the auditing scenario. So if we're going to set the header, always use the `utcNow` from above.
```suggestion
                        message.Headers[Headers.DeliverAt] = DateTimeOffsetHelper.ToWireFormattedString(utcNow.Add(timeDelay));
```

Reply:
I removed the parsing of the TimeSent header and now it's just using the DateTimeOffset.UtcNow